pkg/controllers/dynakube/extension/otel/statefulset_test.go

```diff
@@ -296,8 +317,9 @@ func TestAnnotations(t *testing.T) {
 
 		assert.Len(t, statefulSet.ObjectMeta.Annotations, 1)
 		assert.Empty(t, statefulSet.ObjectMeta.Annotations["a"])
-		assert.Len(t, statefulSet.Spec.Template.ObjectMeta.Annotations, 1)
-		assert.Equal(t, customAnnotations, statefulSet.Spec.Template.ObjectMeta.Annotations)
+		assert.Len(t, statefulSet.Spec.Template.ObjectMeta.Annotations, 2)
```

Comment:
please use `require.Len` to check existence and length of Annotations map and then `assert.<Condition>` to safely verify the contents of the map

Reply:
Fixed!